Plugins/AWSLambdaDeployer/Plugin.swift

```diff
@@ -0,0 +1,433 @@
+            print(output)
+        }
+    }
+    
+    private func generateDeploymentDescriptor(projectDirectory: Path,
+                                              buildConfiguration: PackageManager.BuildConfiguration,
+                                              swiftExecutable: Path,
+                                              samDeploymentDescriptorFilePath: String,
+                                              archivePath: String,
+                                              force: Bool,
+                                              verboseLogging: Bool) throws {
+        print("-------------------------------------------------------------------------")
+        print("Generating SAM deployment descriptor")
+        print("-------------------------------------------------------------------------")
+        
+        //
+        // Build and run the Deploy.swift package description
+        // this generates the SAM deployment descriptor
+        //
+        let deploymentDescriptorFileName = "Deploy.swift"
+        let deploymentDescriptorFilePath = "\(projectDirectory)/\(deploymentDescriptorFileName)"
+        let sharedLibraryName = "AWSLambdaDeploymentDescriptor" // provided by the swift lambda runtime
+        
+        // Check if Deploy.swift exists. Stop when it does not exist.
+        guard FileManager.default.fileExists(atPath: deploymentDescriptorFilePath) else {
+            print("`Deploy.Swift` file not found in directory \(projectDirectory)")
+            throw DeployerPluginError.deployswiftDoesNotExist
+        }
+        
+        do {
+            let cmd = [
+                swiftExecutable.string,
+                "-L \(projectDirectory)/.build/\(buildConfiguration)/",
+                "-I \(projectDirectory)/.build/\(buildConfiguration)/",
+                "-l\(sharedLibraryName)",
+                deploymentDescriptorFilePath,
+                "--archive-path", archivePath
+            ]
+            let helperCmd = cmd.joined(separator: " \\\n")
+            
+            if verboseLogging {
+                print("-------------------------------------------------------------------------")
+                print("Swift compile and run Deploy.swift")
+                print("-------------------------------------------------------------------------")
+                print("Swift command:\n\n\(helperCmd)\n")
+            }
+            
+            // create and execute a plugin helper to run the "swift" command
+            let helperFilePath = "\(projectDirectory)/compile.sh"
+            FileManager.default.createFile(atPath: helperFilePath,
+                                           contents: helperCmd.data(using: .utf8),
+                                           attributes: [.posixPermissions: 0o755])
+            let samDeploymentDescriptor = try Utils.execute(
+                executable: Path("/bin/bash"),
```

Comment:
maybe fine, but we are making assumption that bash is available

Reply:
I added this to dynamically search for ZSH.
ZSH is the default on macOS and available  on ubuntu 
```
        let shellExecutablePath = try self.findExecutable(context: context,
                                                          executableName: "zsh",
                                                          helpMessage: "Zsh is required to run this plugin",
                                                          verboseLogging: configuration.verboseLogging)
```

and later 

```
            let samDeploymentDescriptor = try Utils.execute(
                executable: shellExecutable,
                arguments: ["-c", helperFilePath],
                customWorkingDirectory: projectDirectory,
                logLevel: verboseLogging ? .debug : .silent)

```

Reply:
Q: Do you need a specific shell or could maybe just `/bin/sh` be enough?

Reply:
Just tested with /bin/sh and it works both on macOS and ALI. Thanks